### DISCUSSION TITLE
Functions calling with AgentGroupedChat?

### DISCUSSION THREAD

**Discussion by unsafePtr** [microsoft/semantic-kernel#8319] [Q&A] [answered]
I am importing functions using OpenAPI plugin. SemanticKernel is able to figure out itself when using a single agent. However, when I try to use `AgentGroupChat` it seems it doesn't use or doesn't see available functions. I am using same Kernel across agents. I've checked with ApplicationInsghts and seems there is no `functions` property within the body when sending request.

**Comment by moonbox3** [comment]
Hi @unsafePtr, are you using Python or .Net?

**Reply by unsafePtr** [reply]
.net. I am sorry, can't edit labels after question is created

**Comment by moonbox3** [comment]
@unsafePtr, if using Python, I've got a bug fix related to using plugins with a chat completion agent and an OpenAI assistant agent (#8330) -- but I am not sure if this is the same issue you're running into. I was able to run function calling, and I notice a break occurring when adding the chat completion agent message to the openai assistant agent channel (this is what the PR is fixing).

I've added a sample with plugins -- you'll want to make sure you're execution settings with `FunctionChoiceBehavior` is configured and passed into the chat completion agent. 

If you're using dotnet, then disregard this. :)

**Comment by crickman** [comment, marked as answer]
For `ChatCompletionAgent` please be sure to set execution settings for _auto-invocation_:

https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/GettingStartedWithAgents/Step2_Plugins.cs#L29

If this doesn't help, can you please provide more details as to which agents you are using?

Note: I'd expect the same behavior when using `AgentGroupChat` vs. invoking the agent directly (via `InvokeAsync`)

**Reply by unsafePtr** [reply]
Hi. I am sorry for late answer, as I was on holiday. Can confirm it worked. However, when there are several functions to be called in a sequence I get sometimes exception `The stream was already consumed.`. Maybe there is a specific way to ask user for input within multi-agent chat
